src/linear_solver_logs/solve_log_full_ma.jl

```diff
@@ -0,0 +1,248 @@
+    full residual.
+
+# Fields
+- `collection_rate::Int64`, the frequency with which to record information to append to the
+    remaining fields, starting with the initialization (i.e., iteration 0).
+- `ma_info::MAInfo`, structure that holds information relevant only to moving average, 
+   like the two choices of moving average widths (lambda1 and lambda2), the current moving average
+   width (lambda), a flag to indicate which moving average regime, between lambda1 and lambda2, is 
+   being used, a vector containing the values to be averaged (res_window), and an indicator of where 
+   the next value should be placed (idx).
+- `resid_hist::Vector{Float64}`, a structure that contains the moving average of the error proxy, 
+   typically the norm of residual or gradient of normal equations.
+- `iota_hist::Vector{Float64}`, a structure that contains the moving average of the error proxy, 
+   typically the norm of residual or gradient of normal equations.
+- `width_hist::Vector{Int64}`, data structure that contains the widths used in the moving average 
+   calculation at each iteration.
+- `resid_norm::Function`, a function that accepts a single vector argument and returns a
+    scalar. Used to compute the residual size.
+- `iterations::Int64`, the number of iterations of the solver.
+- `converged::Bool`, a flag to indicate whether the system has converged by some measure.
+- `sampler::Union{DataType,Nothing}`, a data type that is needed for computing constants used in the uncertainty
+  quantification and stopping steps. This is updated with each `log_update!` call. 
+- `max_dimension::Int64`, a value that stores the max between the row and column dimension needed for
+  computation of stopping criterion and uncertainty sets.
+- `sigma2::Union{Float64, Nothing}`, a value that stores the sigma^2 parameter of a sub-Exponential
+  distribution used for determining stopping criterion and uncertainty sets.
+- `omega::Union{Float64, Nothing}`, a value that stores the omega parameter of a sub-Exponential
+  distribution used for determining stopping criterion and uncertainty sets.
+- `eta::Float64, a parameter that allows the adjustment of the uncertainty quantification if the 
+  size of the default covariance is too wide for the particular problem.
+- `true_res`, a boolean indicating if we want the true residual computed instead of approximate.
+# Constructors
+- Calling `LSLogFullMA()` sets `collection_rate = 1`,  `lambda1 = 1`,
+    `lambda2 = 30`, `resid_hist = Float64[]`, `iota_hist = Float64[]`, `width_hist = Int64[]`, 
+    `resid_norm = norm` (Euclidean norm), `iterations = -1`, `converged = false`, 
+    `sampler = LinSysVecRowDetermCyclic`, `max_dimension = 0`, `sigma2 = nothing`, `omega = nothing`,
+    `eta = 1`, and `true_res = false`. The user can specify their own values of lambda1, lambda2, sigma2, omega, and eta using 
+    key word arguments as inputs into the constructor.
+"""
+mutable struct LSLogFullMA <: LinSysSolverLog
```

Comment:
It is not clear to me that all of these fields are needed. Is some of this information not stored in MAInfo or can be determined from MAInfo?

Reply:
I added an extra type called dist_info to reduce the number of fields.